runtime/domain.c

```diff
@@ -1948,9 +1944,8 @@ CAMLexport int caml_bt_is_self(void)
 CAMLexport intnat caml_domain_is_multicore (void)
 {
   dom_internal *self = domain_self;
-  uintnat backup_thread_running =
-    atomic_load_acquire(&self->backup_thread_running);
-  return (!caml_domain_alone() || backup_thread_running);
+  bool bt_running = backup_thread_running(self);
+  return (!caml_domain_alone() || bt_running);
```

Comment:
This is because the name of the function is misleading. What it really tests for is "are there more than one thread running?". So if your program spawned N domains, and all these N domains have exited, `caml_domain_alone` will return `true`, but the backup thread for the main domain has nevertheless been started in the meantime, and is probably still running, hence the check for it.

Reply:
Thanks! My question, more precisely, was why we are not taking advantage of the shortcut behavior of the `||` by writing `!caml_domain_alone || backup_thread_running(self)`.

Reply:
Ah!

The reason is that, prior to this PR, folding the backup thread check into the `if` would make it no longer fit in a single line. With your new wrapper routine, it ought to fit, so go for it!

Reply:
Thanks, I made this small additional change.